### PR TITLE
[Viewer] Only update order balance if it isn't filtered out

### PR DESCRIPTION
**Pull request by nlordell**
This PR updates an order balance only if it is considered valid (i.e. matches the token filter or is valid for the specified batch) for some super awesome gas savings.

### Test Plan

Deployed on Rinkeby and was able to go through [7 pages of unfiltered orders](https://dashboard.tenderly.dev/nlordell/project/simulator/781ea834-e198-41c2-935f-9e771dfb885f) compared to [3 pages of unfiltered orders](https://dashboard.tenderly.dev/nlordell/project/simulator/a86f88e6-cd04-43bf-9ca3-3e07658f670b). We identify calls to `getEncodedUserOrders` by calls to `getUsersPaginated` since it is only used there.

### CODE REVIEW

**Comment by nlordell** [comment]
This might cause problems though, since retrieving "full" pages (i.e. pages with many orders that are considered valid) is considerably more expensive than "empty" pages (i.e. pages with few orders that are considered valid) and hence, the gas check that we currently perform might not be good enough.

**Review by bh2smith — APPROVED** [review]
Short and sweet, wins the race! Super awesome gas savings you have there.

**Comment by fleupold** [comment]
> This might cause problems though, since retrieving "full" pages (i.e. pages with many orders that are considered valid) is considerably more expensive than "empty" pages (i.e. pages with few orders that are considered valid) and hence, the gas check that we currently perform might not be good enough.

Yes, I think this is quite worrying as it makes our already brittle logic there even harder to reason about. I would like to see us move to a fixed "worst-case gas required for next loop" logic (maybe with a factor for page size), before landing this change.

I do agree the gas savings are likely going to be awesome.

**Comment by nlordell** [comment]
> "worst-case gas required for next loop"

Time to tenderly again! Woohoo!

> before landing this change.

I also agree, I'll leave the PR open for now until we implement something like that.

**Comment by nlordell** [comment]
Created #678 and placed this PR on hold.

**Pull request closed without merging**
